state/protocol/badger/state.go

```diff
@@ -357,9 +357,13 @@ func (state *State) bootstrapEpoch(epochs protocol.EpochQuery, segment *flow.Sea
 			if err := verifyEpochSetup(setup, verifyNetworkAddress); err != nil {
 				return fmt.Errorf("invalid setup: %w", err)
 			}
-
 			if err := isValidEpochCommit(commit, setup); err != nil {
-				return fmt.Errorf("invalid commit")
+				return fmt.Errorf("invalid commit: %w", err)
+			}
+
+			err = indexFirstHeight(previous)(tx.DBTxn)
+			if err != nil {
+				return fmt.Errorf("could not index epoch first height: %w", err)
 			}
 
```

Comment:
It seems we have two large chunks of duplicated code, only differing by error comments. 
* [storing previous epoch](https://github.com/onflow/flow-go/blob/a3f746881bf1aa7ced07e9527c3532b3d81d3e15/state/protocol/badger/state.go#L347-L372)
* [storing current epoch](https://github.com/onflow/flow-go/blob/a3f746881bf1aa7ced07e9527c3532b3d81d3e15/state/protocol/badger/state.go#L377-L402)

In addition, we have [this logic](https://github.com/onflow/flow-go/blob/a3f746881bf1aa7ced07e9527c3532b3d81d3e15/state/protocol/badger/state.go#L407-L428) for storing the next epoch, which also has a large overlap. I feel the only relevant difference is that epoch commit event is optional here. 


maybe it is worth-while to encapsulate this logic into an auxiliary function (?). The auxiliary function could take a parameter like "requireEpochCommit" and return epoch setup, epoch commit and EventIDs. 

purely optional suggestion. If that turns out to be complicated (or you feel unnecessarily obfuscates important business logic), just skip this.